bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/discovery/ShellyDiscoveryParticipant.java

```diff
@@ -162,13 +174,13 @@ public DiscoveryResult createResult(@Nullable ServiceInfo service) {
                 addProperty(properties, PROPERTY_DEV_TYPE, thingType);
                 addProperty(properties, PROPERTY_DEV_MODE, mode);
 
-                logger.debug("Adding Shelly thing, UID={}", thingUID.getAsString());
+                logger.debug("{}: Adding Shelly thing, UID={}", name, thingUID.getAsString());
                 return DiscoveryResultBuilder.create(thingUID).withProperties(properties)
                         .withLabel(name + " - " + address).withRepresentationProperty(name).build();
             }
-        } catch (NullPointerException e) {
-            logger.warn("Device discovery failed for device {}, IP {}, service={}: {}", name, address, name,
-                    e.getMessage(), e);
+        } catch (IllegalArgumentException | NullPointerException e) { // maybe some format description was buggy
+            logger.info("{}", messages.get("discovery.failed", name, address, getString(e)));
+            logger.debug("{}: Exception {}\n{}", name, e.getClass(), e.getStackTrace());
```

Comment:
Don't call e.getStacktrace() as part of logging. Just use the exception as the last argument instead.
See other comments for examples.

Reply:
removed all references to e.getStackTrace()